server/src/testFixtures/java/io/crate/types/DataTypeTestCase.java

```diff
@@ -100,15 +106,6 @@ public void test_doc_values_write_and_read_roundtrip_inclusive_doc_mapper_parse(
             writer.addDocument(doc.doc().getFields());
             writer.commit();
 
-            // going through the document mapper must create the same fields
-            ParsedDocument parsedDocument = mapperService.documentMapper().parse(new SourceToParse(
-                table.ident().indexNameOrAlias(),
-                doc.id(),
-                doc.source(),
-                XContentType.JSON
-            ));
-            assertThat(parsedDocument).hasSameFieldsWithNameAs(doc, reference.storageIdent());
-
```

Comment:
If I understand correctly, changing all data type tests from extending `ESTestCase` to `DataTypeTestCase` already makes sure that `DocumentParser` logic is checked in all of them if we keep that code here.

Do I get it right that purpose of introducing dedicated `test_translog_streaming_roundtrip` and moving that logic there has other goal then testing translog?

Reply:
This test is currently doing two things: checking translog roundtrip behaviour, and verifying that docvalues are written correctly.  The idea here is to separate these two checks into separate tests, so that even if a datatype doesn't conform to the docvalues test (eg object type, or the upcoming nested array type) we still run the translog test.